src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1120,18 +1121,43 @@ public Collection<Object> createComponents(
 
         final CompatConfig compatConfig = new CompatConfig(environment, transportPassiveAuthSetting);
 
-        evaluator = new PrivilegesEvaluator(
-            clusterService,
-            threadPool,
-            cr,
-            resolver,
-            auditLog,
-            settings,
-            privilegesInterceptor,
-            cih,
-            irr,
-            namedXContentRegistry.get()
-        );
+        if (PrivilegesEvaluator.USE_LEGACY_PRIVILEGE_EVALUATOR.get(settings)) {
```

Comment:
Nice! Thank you for introducing this setting for rollout. Basically anything referencing this setting will be yanked out to complete https://github.com/opensearch-project/security/issues/5013 when its time to remove this setting to get the full benefits of Optimized Privilege Evaluation?

Reply:
Yes. Bascially, the code to be yanked out exactly corresponds to these commits:

- https://github.com/opensearch-project/security/pull/4998/commits/cc1d22ad795ff6c9597f95c6f9d219edfe0b1823
- https://github.com/opensearch-project/security/pull/4998/commits/310b77ddcfe871706771418b6c9f985515ced149
- https://github.com/opensearch-project/security/pull/4998/commits/ec08e4d9fbc9ea782349d19cb136bba351ebe4b1